card/mod/05_standard/set/type/file.rb

```diff
@@ -1,5 +1,15 @@
 attachment :file, :uploader=>FileUploader
 
+module SelectedAction
+  def select_action_by_params params
+    # test whether request is for current action using card content
+    # if so, we can skip the action lookup
+    rev_id = params[:rev_id]
+    super unless rev_id && db_content =~ /\/#{rev_id}\./
+  end
+end
+include SelectedAction
```

Comment:
right now, most file requests send a rev_id, and we end up looking in the database for that rev_id.  But there's no need to do that, since it's right there in the db content.

I'm not sure this is the best implementation of a fix (feels a bit hacky), but it appears to save a ton of lookups.

At present the machine output is doing a lot of these same kinds of extra lookups.  I think if we fixed that we would be able to load, for example, the home page of a default wagn -- and all associated images -- with zero queries once the cards are cached.  Would be very good to get that kind of page down under 100ms on most servers.